express_relay/programs/express_relay/src/utils.rs

```diff
@@ -96,72 +89,67 @@ pub fn handle_bid_payment(ctx: Context<SubmitBid>, bid_amount: u64) -> Result<()
+    if router_config_account_info.data_len() > 0 {
+        let account_data = &mut &**router_config_account_info.try_borrow_data()?;
+        let router_config_data = ConfigRouter::try_deserialize(account_data)?;
+        split_router = router_config_data.split;
     } else {
-        split_protocol = split_protocol_default;
+        split_router = split_router_default;
     }
 
-    let fee_protocol = bid_amount * split_protocol / FEE_SPLIT_PRECISION;
-    if fee_protocol > bid_amount {
-        // this error should never be reached due to fee split checks, but kept as a matter of defensive programming
-        return err!(ErrorCode::FeesHigherThanBid);
-    }
-
-    let fee_relayer = bid_amount.saturating_sub(fee_protocol) * split_relayer / FEE_SPLIT_PRECISION;
-    if fee_relayer.checked_add(fee_protocol).ok_or(ProgramError::ArithmeticOverflow)? > bid_amount {
+    let fee_router = bid_amount * split_router / FEE_SPLIT_PRECISION;
+    if fee_router > bid_amount {
         // this error should never be reached due to fee split checks, but kept as a matter of defensive programming
         return err!(ErrorCode::FeesHigherThanBid);
     }
+    if fee_router > 0 {
+        let router = &ctx.accounts.router;
+        let balance_router = router.lamports();
+        let rent_router = Rent::get()?.minimum_balance(router.to_account_info().data_len());
+        if balance_router+fee_router < rent_router {
+            return err!(ErrorCode::InsufficientRouterRent);
+        }
 
-    let protocol = &ctx.accounts.protocol;
-    let fee_receiver_protocol = &ctx.accounts.fee_receiver_protocol;
-    if protocol.executable {
-        // validate the protocol fee receiver address as pda if protocol is a program
-        validate_pda(fee_receiver_protocol.key, protocol.key, &[SEED_EXPRESS_RELAY_FEES])?;
-    } else {
-        // if protocol is non-executable pubkey, protocol fee receiver address should = protocol address
-        assert_eq!(protocol.key, fee_receiver_protocol.key);
+        transfer_lamports_cpi(
+            &searcher.to_account_info(),
+            &router.to_account_info(),
+            fee_router,
+            ctx.accounts.system_program.to_account_info()
+        )?;
     }
 
-    let balance_fee_receiver_protocol = fee_receiver_protocol.lamports();
-    let rent_fee_receiver_protocol = Rent::get()?.minimum_balance(0);
-    if balance_fee_receiver_protocol+fee_protocol < rent_fee_receiver_protocol {
-        return err!(ErrorCode::InsufficientProtocolFeeReceiverRent);
+    let fee_relayer = bid_amount.saturating_sub(fee_router) * split_relayer / FEE_SPLIT_PRECISION;
+    if fee_relayer.checked_add(fee_router).ok_or(ProgramError::ArithmeticOverflow)? > bid_amount {
+        // this error should never be reached due to fee split checks, but kept as a matter of defensive programming
+        return err!(ErrorCode::FeesHigherThanBid);
     }
+    if fee_relayer > 0 {
+        let fee_receiver_relayer = &ctx.accounts.fee_receiver_relayer;
```

Comment:
the next 4 lines are somewhat duplicate. We can extract a function to check if fee > rent for any account